### PR TITLE
second try - factorial use derive macro

### PR DESCRIPTION
**Pull request by stevelr**
Signed-off-by: stevelr <steve@cosmonic.com>

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
@stevelr not a comment on the code, I think we should wait until #74 is merged so this can run the build/test CI

**Review by autodidaddict — APPROVED** [review]
I love it when PRs delete more code than they add

**Pull request merged**
